spec/tom_queue/deferred_work/deferred_work_manager_integration_spec.rb

```diff
@@ -30,9 +31,17 @@ def notify(exception)
 
   before(:each) do
     sleep 0.5
+    #binding.pry
 
     Process.kill("SIGTERM", pid)
+    puts "#####"
+    puts "TEST PROCESS ID: #{$$}"
+    puts "$?: #{$?}"
+    puts "DEFERRED (CHILD) PROCESS ID: #{pid}"
+    #binding.pry
     _, @status = Process.waitpid2(pid)
+    puts "WAIT COMPLETE"
+    puts "#####"
   end
 
   it "handles SIGTERM send by god properly" do
```

Comment:
These two tests were failing because we were firing off a `SIGTERM`, but it was being swallowed by `Delayed::Worker` and not being propagated to this deferred worker. To get the tests passing I whacked `Delayed::Worker.raise_signal_exceptions = true` on line 22, which then propagates the SignalException, but doesn't feel right to me.

Why would there be a worker running in this test anyway? Could it be leaking from another test? One to chat through with @joshado

Reply:
Oh! I had these tests failing too, was on my list to look at them. Happy to have a look myself or pair up on it.